viz_scripts/plots.py

```diff
@@ -48,82 +56,125 @@ def merge_small_entries(labels, values):
     disp.display(v2l_df)
 
-    return (v2l_df.index.to_list(),v2l_df.vals.to_list())
-
-
-def format_pct(pct, values):
-    total = sum(values)
-    absolute = int(round(pct*total/100.0))
-    return "{:.1f}%\n({:d})".format(pct, absolute) if pct > 4 else''
+    return (v2l_df.index.to_list(),v2l_df.vals.to_list(), v2l_df.pct.to_list())
 
+def process_data_frame(df, df_col, trip_type):
+    """ Inputs:
+    df = Likely expanded_ct, data_eb or expanded_ct_sensed data frame
+    df_col = Column from the above df, likely Mode_confirm, primary_mode
+    trip_type = Bar labels (e.g. Labeled by user (Confirmed trips))
+    """
+    labels = df[df_col].value_counts(dropna=True).keys().tolist()
+    values = df[df_col].value_counts(dropna=True).tolist()
 
-def pie_chart_mode(plot_title,labels,values,colors_map,file_name):
-    fig, ax = plt.subplots(figsize=(10, 10), subplot_kw=dict(aspect="equal"))
-    m_labels, m_values = merge_small_entries(labels, values)
-
-    wedges, texts, autotexts = ax.pie(m_values,
-                                      labels = m_labels,
-                                      colors=[colors_map[key] for key in labels],
-                                      pctdistance=0.75,
-                                      autopct= lambda pct: format_pct(pct, values),
-                                      textprops={'size': 23})
+    return process_trip_data(labels, values, trip_type)
 
-    ax.set_title(plot_title, size=25)
-    plt.text(-1.3,-1.3,f"Last updated {arrow.get()}", fontsize=10)
-    plt.setp(autotexts, **{'color':'white', 'weight':'bold', 'fontsize':20})
-    plt.savefig(SAVE_DIR+file_name+".png", bbox_inches='tight')
-    plt.show()
+def process_distance_data(df, df_col, distance_col, label_units_lower, trip_type):
+    """ Inputs:
+    df = Likely expanded_ct, data_eb or expanded_ct_sensed data frame
+    df_col = Column from the above df, likely Mode_confirm, primary_mode
+    distance_col = Column associated with distance from above data frame
+    label_units_lower = lbs/kg
+    trip_type = Bar labels (e.g. Labeled by user (Confirmed trips))
+    """
+    dist = df.groupby(df_col).agg({distance_col: ['sum', 'count', 'mean']})
+    dist.columns = ['Total (' + label_units_lower + ')', 'Count', 'Average (' + label_units_lower + ')']
+    dist = dist.reset_index()
+    dist = dist.sort_values(by=['Total (' + label_units_lower + ')'], ascending=False)
 
-def pie_chart_sensed_mode(plot_title,labels,values,file_name):
-    all_labels= ['IN_VEHICLE',
-                 'UNKNOWN',
-                 'WALKING',
-                 'AIR_OR_HSR',
-                 'BICYCLING',
-                 'OTHER']
+    dist_dict = dict(zip(dist[df_col], dist['Total (' + label_units_lower + ')']))
+    labels_dist = []
+    values_dist = []
 
-    val2labeldf = pd.DataFrame({"labels": labels, "values": values})
+    for x, y in dist_dict.items():
+        labels_dist.append(x)
+        values_dist.append(y)
```

Comment:
I have just pulled out the common functionality from the current implementation for `Miles per chosen transport mode` chart in `generic_metrics` and `mode_specific_metrics` notebook into this function `process_distance_data` so it can be re-utilized in `mode_specific_metrics` notebook too.

Reply:
Ah I see. It would have been helpful to clarify that. I still think that this code is ugly, but I agree that the cleanup can be deferred until a future cleanup.